hathor/indexes.py

```diff
@@ -44,57 +44,72 @@ class IndexesManager:
         so it will know which index is better to use in each moment
     """
 
-    def __init__(self, with_tips_index: bool = True) -> None:
-        self.tips_index = TipsIndex() if with_tips_index else None
-        self.txs_index = TransactionsIndex()
+    def __init__(self) -> None:
+        self.all_tips = TipsIndex()
+        self.block_tips = TipsIndex()
+        self.tx_tips = TipsIndex()
+
+        self.sorted_all = TimestampIndex()
+        self.sorted_blocks = TimestampIndex()
+        self.sorted_txs = TimestampIndex()
+
+        self.addresses: Optional[AddressesIndex] = None
+        self.tokens: Optional[TokensIndex] = None
```

Comment:
I think all of these should also be here that are currently in `TransactionStorage`:

```
# Initialize cache for genesis transactions.
self._genesis_cache: Dict[bytes, BaseTransaction] = {}

# Cache of block hash by height
self._block_height_index = BlockHeightIndex()

# Direct and reverse dependency mapping (i.e. needs and needed by)
self._dir_dep_index: Dict[bytes, _DirDepValue] = {}
self._rev_dep_index: Dict[bytes, Set[bytes]] = {}
self._txs_with_deps_ready: Set[bytes] = set()

# Needed txs (key: tx missing, value: requested by)
self._needed_txs_index: Dict[bytes, Tuple[int, bytes]] = {}

# Hold txs that have not been confirmed
self._mempool_tips_index: Set[bytes] = set()

# Hold blocks that can be used as the next parent block
# XXX: if there is more than one they must all have the same score, must always have at least one hash
self._parent_blocks_index: Set[bytes] = {BLOCK_GENESIS.hash}
```

Reply:
Nice catch! I'll move these indexes in a separate PR.

Reply:
I'll mark as resolved and it will be executed soon.